File: dataset/human/python/file_9215.py
import sys 
sys.setrecursionlimit(10**6)
s = input()
def dfs(s):
    while True:
        f = True
        for i in ['dream', 'dreamer', 'erase', 'eraser']:
            if s[-len(i):]==i:
                s = s[:-len(i)]
                if s=='':
                    return True
                f = False
        if f: return False
    
if dfs(s):
    print('YES')
else:
    print('NO')

File: dataset/human/cpp/file_9215.cpp
#include<iostream>
#include<string>
using namespace std;

int main()
{
	int n,m;
	string a[50],b[50];
	cin>>n>>m;
	for(int i=0;i<n;i++) cin>>a[i];
	for(int i=0;i<m;i++) cin>>b[i];
	
	for(int i=0;i<n;i++){
		for(int j=0;j<n;j++){
			int k;
			for(k=0;k<m;k++){
				int l;
				for(l=0;l<m;l++){
					if(a[i+k][j+l]!=b[k][l]) break;
				}
				if(l!=m) break;
			}
			if(k==m){
				cout<<"Yes"<<endl;
				return 0;
			}
		}
	}
	cout<<"No"<<endl;
	return 0;
}
